src/java/org/apache/cassandra/service/accord/AccordService.java

```diff
@@ -240,6 +242,12 @@ private static RuntimeException throwTimeout(TxnId txnId, Txn txn, ConsistencyLe
                             : new ReadTimeoutException(consistencyLevel, 0, 0, false, txnId.toString());
     }
 
+    private static RuntimeException throwPreempted(TxnId txnId, Txn txn, ConsistencyLevel consistencyLevel)
+    {
+        throw txn.isWrite() ? new WritePreemptedException(WriteType.TRANSACTION, consistencyLevel, 0, 0, txnId.toString())
```

Comment:
when you rebase this will become a regression, can you use `CAS`?  I document why in CASSANDRA-18299; `TRANSACTION` requires driver/protocol changes